iceoryx_posh/include/iceoryx_posh/capro/service_description.hpp

```diff
@@ -103,28 +98,23 @@ class ServiceDescription
 
     /// @brief default C'tor
     ServiceDescription() noexcept;
-    ServiceDescription(const ServiceDescription&) = default;
-    ServiceDescription(ServiceDescription&&) = default;
-    ~ServiceDescription() = default;
-
-    /// @brief construction of the capro service description using integers to create a service service description
-    ServiceDescription(uint16_t f_serviceID, uint16_t f_instanceID) noexcept;
+    ServiceDescription(const ServiceDescription&) noexcept = default;
+    ServiceDescription(ServiceDescription&&) noexcept = default;
+    ~ServiceDescription() noexcept = default;
 
     /// @brief construction of the capro service description using fixed strings to create a service service description
+    /// @todo remove
     ServiceDescription(const IdString_t& f_service, const IdString_t& f_instance) noexcept;
 
-    /// @brief construction of the capro service description using integers to create an event service description
-    ServiceDescription(uint16_t f_serviceID, uint16_t f_eventID, uint16_t f_instanceID) noexcept;
-
     /// @brief construction of the capro service description using fixed strings to create an event service description
     ServiceDescription(const IdString_t& f_service,
                        const IdString_t& f_instance,
                        const IdString_t& f_event,
                        ClassHash m_classHash = {0u, 0u, 0u, 0u},
                        Interfaces interfaceSource = Interfaces::INTERNAL) noexcept;
 
-    /// @brief compare operator. If wildcards AnyService, AnyInstance or AnyEvent are used as integer IDs, the
-    /// corresponding member comparisons are skipped. Otherwise, both the integer and the string members are compared.
+    /// @brief compare operator. If wildcards AnyServiceString, AnyInstanceString or AnyEventString are used, the
```

Comment:
I believe wildcards do not belong in a **concrete** service description. They belong into a search pattern of a search. As soon as wildcards are in play, we have an implicit collection of such descriptions (those matching the pattern). Having those is bad design.

This would also make the comparison operator much more lightweight.

Reply:
That's implemented in #854, let's move the discussion over there.